clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -500,11 +520,8 @@ void ConstraintResolver::constrainLocalAssign(Stmt *TSt, DeclaratorDecl *D,
 
   // When the RHS of the assignment is an array initializer, the LHS must be
   // dereferenced in order to generate the correct constraints.
-  bool derefLHS = false;
   if (RHS != nullptr && dyn_cast<InitListExpr>(RHS) != nullptr) {
-    if (D->getType()->isArrayType())
-      derefLHS = true;
-    else {
+    if (!D->getType()->isArrayType()) {
```

Comment:
Why does this check still need to be here? I thought that the `getExpr...` change took care of this?

Reply:
Yeah, it should have been removed. Thanks for catching that.